File: among-eus-core/src/main/java/com/zuehlke/amongeus/core/utility/DistanceCalculator.java
package com.zuehlke.amongeus.core.utility;

import com.zuehlke.amongeus.core.model.Player;

public class DistanceCalculator {

    // Radius of the earth in km
    public static final double RADIUS_OF_WORLD_IN_METER = 6_371_000;

    public static double getDistanceInMeter(Player p1, Player p2){
        var dLat = deg2rad(p1.getLatitude()-p2.getLatitude());  // deg2rad below
        var dLon = deg2rad(p1.getLongitude()-p2.getLongitude());
        var a = Math.sin(dLat/2) * Math.sin(dLat/2) +
                        Math.cos(deg2rad(p1.getLatitude())) * Math.cos(deg2rad(p2.getLatitude())) *
                        Math.sin(dLon/2) * Math.sin(dLon/2);
        var c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a));
        return RADIUS_OF_WORLD_IN_METER * c;
    }

    private static double deg2rad(double deg) {
        return deg * (Math.PI/180);
    }
}

File: players/src/main/kotlin/com/zuehlke/amongeus/players/services/DistanceCalculator.kt
package com.zuehlke.amongeus.players.services

import com.zuehlke.amongeus.players.models.Player

class DistanceCalculator {

    fun getDistanceInKm(p1: Player, p2: Player): Double {
        val R = 6371.0; // Radius of the earth in km
        val dLat = deg2rad(p1.lat-p2.lat)  // deg2rad below
        val dLon = deg2rad(p1.long-p2.long)
        val a = Math.sin(dLat/2) * Math.sin(dLat/2) +
                        Math.cos(deg2rad(p1.lat)) * Math.cos(deg2rad(p2.lat)) *
                        Math.sin(dLon/2) * Math.sin(dLon/2)
        val c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1-a))
        val d = R * c // Distance in km
        return d
    }

    private fun deg2rad(deg:Double):Double {
        return deg * (Math.PI/180)
    }
}